modules/nf-core/hmmer/hmmfetch/main.nf

```diff
@@ -30,7 +30,7 @@ process HMMER_HMMFETCH {
     def keyarg  = key ?: ''
     def kfopt   = keyfile ? '-f' : ''
     def index   = ! key && ! keyfile ? '--index' : ''
-    def outfile = ! key && ! keyfile ? '' : "> ${prefix}.hmm"
+    def outfile = ! key && ! keyfile ? '' : ( "${prefix}.hmm" != hmm ? "> ${prefix}.hmm" : "> ${prefix}.out.hmm" )
```

Comment:
You should not add custom prefixes to the output files. It's better to let the user do that themselves. You can add a check to the module that checks if the input and output file have the same name (many modules already do this)

Reply:
My intention was to add a check, and if that fails, i.e. filenames are identical, write to a different filename. Is this done differently in other modules? Do you know of an example?

Reply:
Hi you can see the guidelines for this here: https://nf-co.re/docs/guidelines/components/modules#command-file-output-naming

There's been a lot of discussion about this lately in the maintainers team but nothing's been decided yet so better to stick with the current guidelines until we've cleared that out